feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts

```diff
@@ -1,84 +1,91 @@
+  > = this.routerData$.pipe(
+    switchMap((routerData) =>
+      this.configCommonsService.getConfiguration(routerData.owner)
+    )
+  );
+
+  currentGroup$: Observable<Configurator.Group> = this.routerData$.pipe(
+    switchMap((routerData) =>
+      this.configuratorGroupsService.getCurrentGroup(routerData.owner)
+    )
+  );
+
+  displayedParentGroup$: Observable<
+    Configurator.Group
+  > = this.configuration$.pipe(
+    switchMap((configuration) =>
+      this.configuratorGroupsService.getMenuParentGroup(configuration.owner)
+    ),
+    switchMap((parentGroup) => this.getCondensedParentGroup(parentGroup))
+  );
+
+  displayedGroups$: Observable<
+    Configurator.Group[]
+  > = this.displayedParentGroup$.pipe(
+    switchMap((parentGroup) => {
+      return this.configuration$.pipe(
+        map((configuration) => {
+          if (parentGroup) {
+            return this.condenseGroups(parentGroup.subGroups);
+          } else {
+            return this.condenseGroups(configuration.groups);
+          }
+        }),
+        delay(0)
+      );
+    })
+  );
 
   iconTypes = ICON_TYPE;
+  GROUPSTATUS = Configurator.GroupStatus;
+
   constructor(
     private configCommonsService: ConfiguratorCommonsService,
     public configuratorGroupsService: ConfiguratorGroupsService,
     private hamburgerMenuService: HamburgerMenuService,
     private configRouterExtractorService: ConfigRouterExtractorService
   ) {}
 
-  GROUPSTATUS = Configurator.GroupStatus;
-
-  ngOnInit(): void {
-    this.configuration$ = this.configRouterExtractorService
-      .extractRouterData()
-      .pipe(
-        switchMap((routerData) =>
-          this.configCommonsService.getConfiguration(routerData.owner)
-        )
-      );
-
-    this.currentGroup$ = this.configRouterExtractorService
-      .extractRouterData()
-      .pipe(
-        switchMap((routerData) =>
-          this.configuratorGroupsService.getCurrentGroup(routerData.owner)
-        )
-      );
-
-    this.displayedParentGroup$ = this.configuration$.pipe(
-      switchMap((configuration) =>
-        this.configuratorGroupsService.getMenuParentGroup(configuration.owner)
-      ),
-      switchMap((parentGroup) => this.getCondensedParentGroup(parentGroup))
-    );
-
-    this.displayedGroups$ = this.displayedParentGroup$.pipe(
-      switchMap((parentGroup) => {
-        return this.configuration$.pipe(
-          map((configuration) => {
-            if (parentGroup) {
-              return this.condenseGroups(parentGroup.subGroups);
-            } else {
-              return this.condenseGroups(configuration.groups);
-            }
-          }),
-          delay(0)
-        );
-      })
-    );
-  }
-
-  clickOnEnter(event, group: Configurator.Group) {
+  /**
+   * Executes a click event to the given group.
+   *
+   * @param event - Keyboard event
+   * @param {Configurator.Group} group - Entered group
+   */
+  clickOnEnter(event, group: Configurator.Group): void {
```

Comment:
Is there a type that we can use for `event`?

Reply:
is KeyboardEvent or better any as a type okay?

Reply:
Please always use a specific type, rather than `any`.
If the `event` can have multiple types, you can specify them like this; `KeyboardEvent | MouseEvent | etc.`

Reply:
Thanks. At the moment we are expecting only a keyboard event. If we expect one more event later, we will add it, as you mentioned above, via pipe operator.